### PR TITLE
feat(colors): update purple colors

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Followed the mappings from https://github.com/dialpad/dialtone/issues/629

- Removed purple-700 and purple-800 color stops.
- Mapped stops as follows (in order):
  - purple-500 -> purple-400
  - purple-600, purple-700 -> purple-500
  - purple-800 -> purple-600
- Updated colors table.
  - Added a new option to have contrast and darkContrast

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [ ] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

**Comment by francisrupert** [comment]
@juliodialpad, 

Awesome start. So many little implications here. I'll go through with fine comb soon. In the meantime, you might start capturing what you've learned that can inform how the [Migration Guide (DT-581)](https://dialpad.atlassian.net/browse/DT-581) is authored,

**Review by francisrupert — COMMENTED** [review]
@juliodialpad,

We'll need to update **images** too. I've updated https://dialpad.atlassian.net/browse/DT-580 to reflect examples to update.

File: docs/_data/colors.json [hunk 1]
```diff
@@ -4,58 +4,45 @@
     "stops": [
       {
         "stop": "100",
-        "hex": "ECE1FF",
+        "hex": "EEE5FF",
         "copy": "dark",
-        "contrast": "AAA 15.58",
+        "contrast": "AAA 7.81",
```
Comment (francisrupert) [inline docs/_data/colors.json:9]: I'm guessing you were using the grid Josh provided to edit these `contrast` values, however they need to be validated and updated. Unfortunately, we have to manually input these; would be nice to have them generated (issue for another day).

For example `Purple 100` is very wrong...

image

Other examples only barely off:

image

image

**Comment by juliodialpad** [comment]
> @juliodialpad,
> 
> We'll need to update **images** too. I've updated https://dialpad.atlassian.net/browse/DT-580 to reflect examples to update.

Nice find! almost forgot about those SVG's, did the same thing, mapped to new stops in order, all svgs using those values should be OK now.

**Comment by dialpad-drew** [comment]
CC: @hynes-dialpad

**Review by francisrupert — COMMENTED** [review]
I've comprehensively gone through the entire codebase again, in addition to these updated files with a fine comb, and I can't imagine we've missed anything. Pending the one outstanding update for **Contrast** values, https://github.com/dialpad/dialtone/pull/633#discussion_r935918173, and after @juliodialpad's last round of commits (images, and other random locations he identified), I **provisionally** approve this to be moved into the `version7` branch.

I'd like to get @hellohynes (or is it @hynes-dialpad) to review to validate. I'd already sent this to him earlier today under separate cover, so in the meantime I'd like at least @braddialpad to eyeball it for any general reaction, and _optionally_ @dialpad-drew.

Awesome work @juliodialpad!

**Review by braddialpad — APPROVED** [review]
I see no problems on the coding side of things here, looks good!

File: docs/_data/colors.json (hunk 1 above)
Comment (francisrupert) [reply docs/_data/colors.json:9]: Trapped this to [DT-597](https://dialpad.atlassian.net/browse/DT-597).

**Pull request merged**